doc/migrations/v0.46-v1.0.md

```diff
@@ -0,0 +1,53 @@
+<!--Specify versions for migration below-->
+# Migrating to libp2p@1.0 <!-- omit in toc -->
+
+A migration guide for refactoring your application code from libp2p v0.46 to v1.0.
+
+## Table of Contents <!-- omit in toc -->
+
+- [API](#api)
+- [Module Updates](#module-updates)
+- [Metrics](#metrics)
+
+## API
+
+<!--Describe breaking APIs with examples for Before and After
```

Comment:
Are these comments necessary?

Reply:
They're from the template

Reply:
I left these comments because other PRs will fill in these sections